framework/src/reporters/PerfGraphReporter.C

```diff
@@ -0,0 +1,95 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PerfGraphReporter.h"
+
+#include "PerfGraph.h"
+
+registerMooseObject("MooseApp", PerfGraphReporter);
+
+InputParameters
+PerfGraphReporter::validParams()
+{
+  InputParameters params = GeneralReporter::validParams();
+  params.addClassDescription("Reports the full performance graph from the PerfGraph.");
+  return params;
+}
+
+PerfGraphReporter::PerfGraphReporter(const InputParameters & parameters)
+  : GeneralReporter(parameters)
+{
+  // We store the value as a PerfGraph * here because we can't use an actual PerfGraph - the
+  // reporter system will try to default construct a PerfGraph. Upon restart/recover, this
+  // the type const PerfGraph * will be loaded as nullptr because it doesn't really make
+  // sense to keep the PerfGraph state (as of yet...). We don't store the reporter value
+  // here because we don't actually need it - the filling action is performed all in the
+  // to_json method for const PerfGraph *
+  declareValueByName<const PerfGraph *>("perf_graph", REPORTER_MODE_DISTRIBUTED) = &_perf_graph;
+}
+
+void
+to_json(nlohmann::json & json, const PerfGraph * const & perf_graph)
+{
+  // Must update the timing before filling so that the data in each
+  // PerfNode is up to date
+  const_cast<PerfGraph *>(perf_graph)->update();
```

Comment:
I'm a little hesitant on putting all this computation in the output function. Perhaps it would be better to have an `execute` function in this reporter and have the reporter value be a `std::vector<PerfGraph::PerfNodeInfo *>`. Then you only need a `to_json` method for `const PerfGraph::PerfNodeInfo * const & info`.  At the least, I think the `update()` call should be in the `execute` function.

Reply:
I vote for what @loganharbour already has here.  The `update` just updates the current times... it should be incredibly quick.

Reply:
Too late - literally just pushed 😆 

This enables proper data store/load serialization as well... not sure if it's _needed_, but it's there.

The other issue is that if you do all of the work in `to_json`, you cannot enable output based on `EXECUTE_ON` with how the reporter system is designed